src/net_processing.cpp

```diff
@@ -3091,12 +3100,12 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             LOCK(cs_main);
             Misbehaving(pfrom->GetId(), 100);
         }
-        else
+        else if (pfrom->m_tx_relay != nullptr)
```

Comment:
Hmm, should you add an else branch here and disconnect them if we opened a blocksonly connection to some peer that asked for filtering?